examples/gocean/eg6/test.x90

```diff
@@ -0,0 +1,34 @@
+Program test
```

Comment:
Rather than duplicate this and the two kernel files, please could you copy what is done in the lfric examples and add a `examples/gocean/code` directory and put them in there.

Reply:
OK, I would actually disagree with that:

1. Even though some files in eg5/6 are identical atm, for #638 I have already modified one version.
2. I looked at lfric, and found e.g. 5 identically named function (one in ``code``), all slightly different, and no obvious way to see where the one in ``code`` is actually used from. So I think I am not the only one who adjusts a test case for various examples :grin:
3. Keeping all examples in sync if one needs modification just makes more work
4. I can't remember where it happened (I believe while looking at nemo test cases) I was actually confused on where I could find the source code - I certainly expected the sources to be 'close'.
4. And imho most important: I actually like to play with a test case (esp. if it is executable): modify it, recompile and see if it works as expected. So I _really_ prefer to have everything self-contained in one directory, so I can do whatever I want, and won't break other examples.

Even if you agree to move the code into ``code`` ... it is quite a bit of work (what do I need to tell psyclone to find kernels in that directory ... I tried ``-I`` and that didn't work) and totally unrelated to this PR ... then I would prefer to open a separate ticket for that and do it later.